crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -189,6 +284,35 @@ impl TransactionsForAccount for PendingTransactionsForAccount {
         // is equal to the account nonce
         self.txs().contains_key(&previous_nonce) || ttx.signed_tx.nonce() == current_account_nonce
     }
+
+    fn has_balance_to_cover(
+        &self,
+        ttx: &TimemarkedTransaction,
+        current_account_balances: &HashMap<IbcPrefixed, u128>,
+    ) -> bool {
+        let mut remaining_account_balances = current_account_balances.clone();
+
+        // deduct costs from current account balances and return false if missing funds for any
+        for tx in self.txs.values().chain(std::iter::once(ttx)) {
```

Comment:
does this work if `ttx` has a nonce that isn't directly sequential to `self.txs`?

Reply:
Yes, the `.chain(x)` appends `x` as the last element in the iterator. It doesn't apply any sorting or deduping logic to the additional element. We only ever trigger this after ensuring that the nonce is sequential but it's not necessary